packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx

```diff
@@ -21,32 +33,108 @@ type ArrowDirection = 'right' | 'left';
   styleUrl: 'atomic-segmented-facet-scrollable.pcss',
   shadow: true,
 })
-export class AtomicSegmentedFacetScrollable {
+export class AtomicSegmentedFacetScrollable implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
   private horizontalScroll?: HTMLDivElement;
+  @State() private hideLeftArrow = true;
+  @State() private hideRightArrow = false;
+  private observer!: ResizeObserver;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  componentDidLoad() {
+    this.observer = new ResizeObserver(() => {
+      this.handleScroll();
+    });
+    this.observer.observe(this.horizontalScroll as HTMLDivElement);
+  }
+
+  disconnectedCallback() {
+    this.observer.disconnect();
+  }
+
+  @Listen('wheel')
+  @Listen('touchmove')
+  handleScroll() {
+    if (!this.horizontalScroll) {
+      return;
+    }
+
+    const isScrollable =
+      this.horizontalScroll.clientWidth < this.horizontalScroll.scrollWidth;
+    const isLeftEdge = Math.floor(this.horizontalScroll.scrollLeft) <= 0;
+    const isRightEdge =
+      Math.ceil(this.horizontalScroll.scrollLeft) >=
+      this.horizontalScroll.scrollWidth - this.horizontalScroll.clientWidth;
+
+    this.hideLeftArrow = !isScrollable || isLeftEdge;
+    this.hideRightArrow = !isScrollable || isRightEdge;
+  }
 
   private slideHorizontally(direction: ArrowDirection) {
     const container = this.horizontalScroll;
-    const pixelsToScroll = container ? container.clientWidth * 0.75 : 700;
-
-    if (container === null || !container) {
+    if (!container) {
       return;
     }
+
+    const width = container.clientWidth;
+    const scrollWidth = container.scrollWidth;
+    const scrollLeft = container.scrollLeft;
```

Comment:
You could get rid of these variables (recommended since they contain no calculations), or use `width` at line 93 for consistency